site-src/geps/gep-724.md

```diff
@@ -0,0 +1,636 @@
+are interested in the more advanced capabilities that some form of selection may
+enable, but at this point it's clear that a large portion of users value an
+explicit way to attach a Route to a Gateway.
+
+#### 6. We Need to Maintain a Handshake Between Gateways and Routes
+Of course we do still need a handshake that will enable cross-namespace
+references between Routes and Gateways. This proposal leaves in the core
+capabilities of the v1alpha1 API for this. Gateways can specify the namespaces
+they trust Routes to bind from, and Routes directly reference the Gateways they
+want to attach to. This is largely similar to the ReferencePolicy model proposed
+for Route->Service references, but is embedded within the Route and Gateway
+resources. The alternatives below explore what this could look like with
+ReferencePolicy.
+
+## API Changes
+
+The proposed changes here can be summarized as:
+
+* Remove Route selector from the RouteBindingSelector in Gateway listeners.
+* Replace Route kind and group with optional list of accepted kinds and groups
+  in RouteBindingSelector.
+* Rename RouteBindingSelector to ListenerRoutes.
+* Replace the 3 options from Route -> Gateway (All, FromList, SameNamespace)
+  with a reference list that supports arbitrary kinds.
+
+### Gateway
+
+On Gateway, the only change involves removing the Route selector field.
+Everything else remains the same.
+
+#### Removed
+
+```go
+  // Selector specifies a set of route labels used for selecting
+  // routes to associate with the Gateway. If this Selector is defined,
+  // only routes matching the Selector are associated with the Gateway.
+  // An empty Selector matches all routes.
+  //
+  // Support: Core
+  //
+  // +optional
+  Selector *metav1.LabelSelector `json:"selector,omitempty"`
+  // Group is the group of the route resource to select. Omitting the value or specifying
+  // the empty string indicates the gateway.networking.k8s.io API group.
+  // For example, use the following to select an HTTPRoute:
+  //
+  // routes:
+  //   kind: HTTPRoute
+  //
+  // Otherwise, if an alternative API group is desired, specify the desired
+  // group:
+  //
+  // routes:
+  //   group: acme.io
+  //   kind: FooRoute
+  //
+  // Support: Core
+  //
+  // +optional
+  // +kubebuilder:default=gateway.networking.k8s.io
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Group *string `json:"group,omitempty"`
+  // Kind is the kind of the route resource to select.
+  //
+  // Kind MUST correspond to kinds of routes that are compatible with the
+  // application protocol specified in the Listener's Protocol field.
+  //
+  // If an implementation does not support or recognize this
+  // resource type, it SHOULD set the "ResolvedRefs" condition to false for
+  // this listener with the "InvalidRoutesRef" reason.
+  //
+  // Support: Core
+  Kind string `json:"kind"`
+```
+
+#### Added
+```go
+type ListenerRoutes struct {
```

Comment:
Nit: this would be set on the listener `Routes` field, which implies the following addition:

```Go
type Listener struct {
...
Routes ListenerRoutes
}
```

Or is this being added to new fields in `RouteSelector`?

Reply:
Yeah this was confusing, I added some text for this + a related change to describe where they would fit in.